### PR TITLE
Fixes for "Can not convert variables for Cloud Storage rules runtime"

### PR DESCRIPTION
**Pull request by abeisgoat**
Adds boolean type to ExpressionValue translation along with other iOS fixes

### CODE REVIEW

**Review by samtstern — APPROVED** [review]

File: src/emulator/storage/rules/runtime.ts [hunk 1]
```diff
@@ -313,6 +313,10 @@ function toExpressionValue(obj: any): ExpressionValue {
     return { string_value: obj };
   }
 
+  if (typeof obj == "boolean") {
```
Comment (yuchenshi) [inline src/emulator/storage/rules/runtime.ts:316]: nit: these should use triple equals `===`.

File: src/emulator/storage/rules/runtime.ts [hunk 2]
```diff
@@ -313,6 +313,10 @@ function toExpressionValue(obj: any): ExpressionValue {
...
+    return { bool_value: obj };
+  }
+
   if (typeof obj == "number") {
```
Comment (yuchenshi) [inline src/emulator/storage/rules/runtime.ts:320]: Ideally we should have some tests for these branches and corner cases (e.g. `undefined`)

**Review by yuchenshi — APPROVED** [review]
